sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -394,15 +409,23 @@ public Mono<List<Address>> getServerAddressesViaGatewayAsync(
         Instant addressCallStartTime = Instant.now();
         HttpRequest httpRequest = new HttpRequest(HttpMethod.GET, targetEndpoint, targetEndpoint.getPort(), httpHeaders);
 
-        Mono<HttpResponse> httpResponseMono;
-        if (tokenProvider.getAuthorizationTokenType() != AuthorizationTokenType.AadToken) {
-            httpResponseMono = this.httpClient.send(httpRequest,
-                Duration.ofSeconds(Configs.getAddressRefreshResponseTimeoutInSeconds()));
-        } else {
+        Duration responseTimeout= Duration.ofSeconds(Configs.getAddressRefreshResponseTimeoutInSeconds());
+        Mono<HttpResponse> httpResponseMono = this.httpClient.send(httpRequest, responseTimeout);
+
+        if (tokenProvider.getAuthorizationTokenType() == AuthorizationTokenType.AadToken) {
             httpResponseMono = tokenProvider
                 .populateAuthorizationHeader(httpHeaders)
-                .flatMap(valueHttpHeaders -> this.httpClient.send(httpRequest,
```

Comment:
why did we remove this? - now are we not sending it for every request header?

Reply:
it is changed to use httpResponseMono

Reply:
revert back to follow original format to reduce confusion